ui/imports/shared/panels/StatusAssetSelector.qml

```diff
@@ -0,0 +1,168 @@
+                    font.pixelSize: 15
+                    width: Math.min(50, implicitWidth)
+                    elide: Text.ElideRight
+                    verticalAlignment: Text.AlignVCenter
+                    color: Theme.palette.directColor1
+                    font.weight: Font.Medium
+                }
+                StatusRoundedImage {
+                    id: iconImg
+                    width: 40
+                    height: 40
+                    image.onStatusChanged: {
+                        if (iconImg.image.status === Image.Error) {
+                            iconImg.image.source = defaultToken
+                        }
+                    }
+                }
+            }
+        }
+        selectMenu.delegate: menuItem
+    }
+
+    Component {
+        id: menuItem
+        MenuItem {
+            id: itemContainer
+            property bool isFirstItem: index === 0
+            property bool isLastItem: index === assets.rowCount() - 1
+
+            width: parent.width
+            height: 72
+            StatusRoundedImage {
+                id: iconImg
+                anchors.left: parent.left
+                anchors.leftMargin: 16
+                anchors.verticalCenter: parent.verticalCenter
+                image.source: root.tokenAssetSourceFn(symbol.toUpperCase())
+                image.onStatusChanged: {
+                    if (iconImg.image.status === Image.Error) {
+                        iconImg.image.source = defaultToken
+                    }
+                }
+            }
+            Column {
+                anchors.left: iconImg.right
+                anchors.leftMargin: 12
+                anchors.verticalCenter: parent.verticalCenter
+
+                StatusBaseText {
+                    text: symbol.toUpperCase()
+                    font.pixelSize: 15
+                    color: Theme.palette.directColor1
+                }
+
+                StatusBaseText {
+                    text: name
+                    color: Theme.palette.baseColor1
+                    font.pixelSize: 15
+                }
+            }
+            Column {
+                anchors.right: parent.right
+                anchors.rightMargin: 16
+                anchors.verticalCenter: parent.verticalCenter
+                StatusBaseText {
+                    font.pixelSize: 15
+                    text: parseFloat(totalBalance).toFixed(4) + " " + symbol
+                }
+                StatusBaseText {
+                    font.pixelSize: 15
+                    anchors.right: parent.right
+                    text: getCurrencyBalanceString(totalCurrencyBalance)
+                    color: Theme.palette.baseColor1
+                }
+            }
+            background: Rectangle {
+                color: itemContainer.highlighted ? Theme.palette.statusSelect.menuItemHoverBackgroundColor : Theme.palette.statusSelect.menuItemBackgroundColor
+            }
+            Component.onCompleted: {
```

Comment:
Ouch... this definitely doesn't belong here, in a Component's background property

Reply:
I can understand that it looks really complicated, honestly was just moving it out of StatusQ and didnt really change much when moving it. In case we think we need to simplify this we can create a separate task for it?

Reply:
Fair enough